lms/static/scripts/frontend_apps/components/LaunchErrorDialog.tsx

```diff
@@ -320,6 +320,42 @@ export default function LaunchErrorDialog({
         </ErrorModal>
       );
 
+    case 'canvas_studio_download_unavailable':
+      return (
+        <ErrorModal
+          {...defaultProps}
+          onRetry={undefined}
+          title="Unable to fetch Canvas Studio video"
+        >
+          <p>
+            Only videos uploaded directly to Canvas Studio can be used. Videos
+            hosted on YouTube or Vimeo cannot be used.
+          </p>
+        </ErrorModal>
+      );
```

Comment:
Something slightly confusing about these error messages is the fact that what you see here is not the entire message. Instead the server-provided message is shown at the top, and the children of the `ErrorModal` are shown underneath.

For some of these errors, the server-side response includes only a code, for others there is a code and a message. Here I followed the pattern of those errors which include a short summary in the response, and then include extra details as `ErrorModal` children.

If we want to translate these error messages in future, then we'll need to have them all in one place though.